File: backend/src/main/java/com/lykawka/collectr/app/model/item/Item.java
package com.lykawka.collectr.app.model.item;

import com.lykawka.collectr.app.model.BaseEntity;
import com.lykawka.collectr.app.model.collection.Collection;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Data
@Entity
@Table(name = "items")
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Item extends BaseEntity {

    @Column(nullable = false)
    private String name;

    private String description;

    private String manufacturer;

    private String imageUrl;

    @Enumerated(EnumType.STRING)
    private ItemCondition condition;

    private Integer releaseYear;

    private Double estimatedValue;

    private Double purchasePrice;

    @Enumerated(EnumType.STRING)
    private ItemSituation situation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "collectionId", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Collection collection;
}

File: frontend/collectr/types/item.ts
export enum ItemCondition {
  MINT = "MINT",
  NEAR_MINT = "NEAR_MINT",
  VERY_GOOD = "VERY_GOOD",
  GOOD = "GOOD",
  ACCEPTABLE = "ACCEPTABLE",
  POOR = "POOR",
  DAMAGED = "DAMAGED",
}

export enum ItemSituation {
  OWNED = "OWNED",
  WISHLIST = "WISHLIST",
  FOR_SALE = "FOR_SALE",
  SOLD = "SOLD",
}

export interface ItemDTO {
  id: string;
  collectionId: string;
  name: string;
  description: string | null;
  imageUrl: string | null;
  manufacturer: string | null;
  releaseYear: number | null;
  estimatedValue: number | null;
  purchasePrice: number | null;
  condition: ItemCondition | null;
  situation: ItemSituation;
  active: boolean;
}

export interface CreateItemRequest {
  name: string;
  description?: string;
  imageUrl?: string;
  manufacturer?: string;
  releaseYear?: number;
  estimatedValue?: number;
  purchasePrice?: number;
  condition?: ItemCondition;
  situation: ItemSituation;
  collectionId: string;
}

export interface UpdateItemRequest {
  name: string;
  description?: string;
  imageUrl?: string;
  manufacturer?: string;
  releaseYear?: number;
  estimatedValue?: number;
  purchasePrice?: number;
  condition?: ItemCondition;
  situation: ItemSituation;
}

export type Item = ItemDTO;
